lib/elixir/lib/kernel.ex

```diff
@@ -2968,6 +2975,23 @@ defmodule Kernel do
 
       when x >= 1 and x <= 3
 
+  ### Special AST Considerations
+
+  `not in` is rewritten in the Elixir compiler, so `:left not in :right` appears
```

Comment:
And it would be clearer to say "`not in` is parsed by the compiler into the `{:not, [], [{:in, _, [left, right]}]}`". There is no such thing as being rewritten, it is just how it is parsed. :)